### PR TITLE
Onboarding python

### PR DESCRIPTION
**Pull request by dvmarcilio**
This PR adds the Python grammar as a dependency and a custom cleanup demo.

## Demo

The demo deletes a given string literal that is a list element. If this string literal is the only element, a custom cleanup will delete an assignment to the resulting empty list (e.g., `deps3` in the example below).
It also replaces a string literal that is a list element with another string literal.

Given that:

- the string literal argument to delete is `dependency2`.
- the string literal to replace is `dependency1` and its replacement is `dependency1_1`.

Before

```python
deps = ['dependency1', 'dependency2', 'dependency3']
nums = [1, 2, 3]
deps2 = ["dependency2", "dependency1"]
deps3 = ['dependency2']
empty_list = []
```

After

```python
deps = ['dependency1_1', 'dependency3']
nums = [1, 2, 3]
deps2 = ["dependency1_1"]
empty_list = []
```

### CODE REVIEW

File: polyglot/piranha/demo/find_replace_custom_cleanup_demos.py [hunk 1]
```diff
@@ -14,5 +14,16 @@ def java_demo():
     print("Running the Find/Replace Custom Cleanup demo for Java")
     _ = run_piranha_cli(join(find_Replace_dir, "java"), join(find_Replace_dir, "java/configurations"), True)
 
+def python_demo():
+    """
+    Deletes the @str_literal (substitution in `piranha_arguments`) string literal when present as list element.
```
Comment (ketkarameya) [inline polyglot/piranha/demo/find_replace_custom_cleanup_demos.py:19]:
```suggestion
    Deletes the string literal `@str_literal` (from `substitutions` in `piranha_arguments.toml`) if it appears as a list element.
```

File: polyglot/piranha/demo/find_replace_custom_cleanup/python/configurations/rules.toml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# After
+#  a = [ 'a', 'c']
+#
+[[rules]]
+name = "delete_str_literal_single_quote_in_list"
+groups = ["delete_str_literal_in_list"]
+query = """
+(
+    (list
+        (string) @s
```
Comment (ketkarameya) [inline polyglot/piranha/demo/find_replace_custom_cleanup/python/configurations/rules.toml:27]: nicer names? say `@string_literal`

File: polyglot/piranha/demo/find_replace_custom_cleanup_demos.py [hunk 3]
```diff
@@ -14,5 +14,16 @@ def java_demo():
     print("Running the Find/Replace Custom Cleanup demo for Java")
     _ = run_piranha_cli(join(find_Replace_dir, "java"), join(find_Replace_dir, "java/configurations"), True)
 
+def python_demo():
```
Comment (ketkarameya) [inline polyglot/piranha/demo/find_replace_custom_cleanup_demos.py:17]: Can we please add another small rule that updates "dependency3" -> "dependency3_1"

**Review by ketkarameya — CHANGES_REQUESTED** [review]
Left a few comments.

File: polyglot/piranha/demo/find_replace_custom_cleanup/python/only_lists.py [hunk 4]
```diff
@@ -0,0 +1,5 @@
+deps = ['dependency1', 'dependency2', 'dependency3']
```
Comment (ketkarameya) [inline polyglot/piranha/demo/find_replace_custom_cleanup/python/only_lists.py:1]: Add license to all new files.

File: polyglot/piranha/test-resources/python/structural_find/input/list_as_arg.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
+func_call(
```
Comment (ketkarameya) [inline polyglot/piranha/test-resources/python/structural_find/input/list_as_arg.py:1]: license

File: polyglot/piranha/test-resources/python/structural_find/input/only_lists.py [hunk 6]
```diff
@@ -0,0 +1,3 @@
+deps = ['dependency1', 'dependency2', 'dependency3']
```
Comment (ketkarameya) [inline polyglot/piranha/test-resources/python/structural_find/input/only_lists.py:1]: license

**Comment by dvmarcilio** [comment]
Thanks for the review and the suggestions. I tried to address all of them.
I updated `dependency1` -> `dependency1_1` so we replace both `'dependency1'` and `"dependency1"` (single and double quotes).

If the changes are good enough, feel free to squash the commits.

**Review by ketkarameya — APPROVED** [review]
LGTM.

**Pull request merged**